VariantSurvival_package/R/VariantSurvival.R

```diff
@@ -1,423 +1,359 @@
-        de2 <- de2[-(1)]
-        de2 <- de2[-(4)]
-        df <- de2 %>%
-          mutate(variant= ifelse(variant=="Yes", 1, 0))
-        #input factors
-        #df <- df %>% rename(Time = Time_to_death_or_last_followup_days)
-        df <- df %>% rename(Time = input$time)
-        df <- df %>% rename(Phenotype = input$phenotype)
-        #
-        s <- survfit(Surv(Time,variant)~ Phenotype, data = df)
-        g <- ggsurvplot(
-          s,
-          conf.int = TRUE,
-          data = df,
-          risk.table = TRUE
-        )
-        g
-        dev.off()
-      })
-    #d plot
-    output$download3plot <- downloadHandler(
-      filename = function(){
-        paste("survival_plot2", "png", sep=".")
-      },
-      content = function(file){
-        png(file)
-        gene_sv2 <- as.data.frame(c(allgenes_sv["patient_ID"], allgenes_sv[input$targetGene]))
-        colnames(gene_sv2) <- c('patient_ID','Structural_Variants_count')
-        gene_sv2$variant <- "No"
-        gene_sv2$variant[gene_sv2$Structural_Variants_count > 1 ] <- "Yes"
-        de2 <- merge(gene_sv2, metadata2, by=0, all=TRUE)
-        de2 <- de2[-(1)]
-        de2 <- de2[-(4)]
-        df <- de2 %>%
-          mutate(variant= ifelse(variant=="Yes", 1, 0))
-        #input factors
-        #• df <- df %>% rename(Time = Time_to_death_or_last_followup_days)
-        df <- df %>% rename(Time = input$time)
-
-        #df <- df %>% rename(Phenotype = "Phenotype")
-        df <- df %>% rename(Phenotype = input$phenotype)
-        #
-        s <- survfit(Surv(Time,variant)~ Phenotype, data = df)
-        g2 <-ggsurvplot_facet(
-          s,
-          conf.int = TRUE,
-          data = df,
-          facet.by = "Structural_Variants_count")
-        g2
-        dev.off()
-      })
-  }
-
-  # Run the application
-  shinyApp(ui = ui, server = server)
-}
+#' VariantSurvival
+#'
+#' @param vcffile path to the vcf file containing the Structural variant data
+#' @param metadatafile path to the txt file containing the samples metadata
+#'
+#' @return
+#' @export
+#'
+
+VariantSurvival <- function(vcffile, metadatafile){
+  install_load_requirements()
+  # parse inputs
+  vcf <- vcfR::read.vcfR(vcffile, verbose = FALSE)
+  metadata <- readxl::read_excel(metadatafile)
+  # remove empty extra lines
+  metadata <- na.omit(metadata)
+  # create user interface layout
+
+
+  ui <- bootstrapPage(
+    navbarPage(theme = shinytheme("flatly"),
+               collapsible = TRUE,
+               HTML('<a style="text-decoration:none;
+               cursor:default;
+                    color:#FFFFFF;
+                    " class="active" href="#">VariantSurvival</a>'),
+               id="nav",
+               windowTitle ="VariantSurvival",
+               sidebarLayout(
+                 sidebarPanel(
+                   pickerInput(inputId ="disease_n",
+                               label = "Diseases",
+                               choices = c("Amyotrophic lateral sclerosis"=1,
```

Comment:
If we can outsource the list of conditions into a file and read them upon the start of the app I think it would increase maintainability in the long term.

Reply:
Which conditions exactly?

Reply:
L:33-39, sorry I am referring to the list of neurological conditions as listed in "choices". It's not too urgent really but I think if the neurological conditions reside somewhere in a separate file we wouldn't need to patch `VariantSurvival.R` if we add diseases.

Reply:
Oh! I see now, you're right but we would still need to patch it for getting the genes of interest no?

Reply:
Affirmative, I suppose it is the `get_disease_gene_list` function that'd need to be generic as well.